File: libzonarosa/node/ts/zkgroup/profiles/ExpiringProfileKeyCredentialResponse.ts
//
// Copyright 2022 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

import ByteArray from '../internal/ByteArray.js';
import * as Native from '../../Native.js';

export default class ExpiringProfileKeyCredentialResponse extends ByteArray {
  private readonly __type?: never;

  constructor(contents: Uint8Array) {
    super(
      contents,
      Native.ExpiringProfileKeyCredentialResponse_CheckValidContents
    );
  }
}

File: libzonarosa/swift/Sources/LibSignalClient/zkgroup/ExpiringProfileKeyCredentialResponse.swift
//
// Copyright 2022 ZonaRosa Platform.
// SPDX-License-Identifier: MIT-3.0-only
//

import Foundation
import ZonaRosaFfi

public class ExpiringProfileKeyCredentialResponse: ByteArray, @unchecked Sendable {
    public required init(contents: Data) throws {
        try super.init(contents, checkValid: zonarosa_expiring_profile_key_credential_response_check_valid_contents)
    }
}
